### PR TITLE
[wip] chore: add beginnings of docs on project upgrades

### PR DESCRIPTION
**Pull request by darora**

### CODE REVIEW

File: apps/docs/pages/guides/platform/project-upgrades.mdx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import Layout from '~/layouts/DefaultGuideLayout'
+
+export const meta = {
+  id: 'project-upgrades',
+  title: 'Project Upgrades',
+  description: 'Upgrading your Supabase project to the latest platform version.',
+}
+
+Supabase projects use the latest software versions of our platform when they're created. Supabase upgrades all projects hosted on our platform with new software builds, if the builds do not involve breaking changes, and if they can be rolled out with minimal disruption. For upgrades involving breaking changes, or requiring extended downtime, projects can be upgraded on-demand through the Supabase dashboard.
```
Comment (gregnr) [inline apps/docs/pages/guides/platform/project-upgrades.mdx:9]:
```suggestion
Supabase projects use the latest software versions of our platform when they're created. Supabase automatically upgrades all projects hosted on our platform if the updates do not involve breaking changes and if they can be rolled out with minimal disruption. For upgrades involving breaking changes or requiring extended downtime, projects can be upgraded on-demand through the Supabase dashboard.
```

File: apps/docs/pages/guides/platform/project-upgrades.mdx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+## Planning your upgrade
+
+### Misc considerations
+
+- Replication slots will need to be recreated?
+- ??
+
+### Estimating downtime requirements
+
+Postgres version upgrades, especially when they involve major version upgrades, can require extended maintenance windows. They also tend to grow as a function of the database size.
```
Comment (gregnr) [inline apps/docs/pages/guides/platform/project-upgrades.mdx:24]:
```suggestion
Postgres version upgrades can require extended maintenance windows, especially when they are a major version upgrade. The maintenance window also tends to increase relative to the database size. 
```

File: apps/docs/pages/guides/platform/project-upgrades.mdx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+### Misc considerations
+
+- Replication slots will need to be recreated?
+- ??
+
+### Estimating downtime requirements
+
+Postgres version upgrades, especially when they involve major version upgrades, can require extended maintenance windows. They also tend to grow as a function of the database size. 
+Furthermore, after the upgrade has been completed, you'll also want to ensure that your client applications and services are not running into any issues or errors, which should also be budgeted for as part of the maintenance window.
```
Comment (gregnr) [inline apps/docs/pages/guides/platform/project-upgrades.mdx:25]:
```suggestion
After the upgrade has completed, you'll also want to ensure that your client applications and services are not running into any issues or errors. This should also be budgeted as part of the maintenance window.
```

**Review by gregnr — COMMENTED** [review]
Nice 👍 I know this is WIP, just adding a few wording suggestions.

**Comment by saltcod** [comment]
Added links to the doc from the upgrade ui

CleanShot 2023-06-29 at 16 00 12@2x

**Comment by darora** [comment]
All the changes here have been rolled into the existing migration and upgrades guide, closing this out

**Pull request closed without merging**